pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -135,6 +135,15 @@ def is_enabled(self) -> bool:
     def is_disabled(self) -> bool:
         return not self.is_enabled
 
+    @property
+    def _val_processed(self):
+        if self.trainer is None:
+            return 0
+        if self.trainer.state.fn == "fit":
+            # use total in case validation runs more than once per training epoch
+            return self.trainer.fit_loop.epoch_loop.val_loop.epoch_loop.batch_progress.total.processed
```

Comment:
does that mean we reset the val_loop batch_progress.total every training epoch? I couldn't find that in the code. That would be a requirement to make this work with the current way the bar is supposed to report progress on mutiple val runs perepoch

Reply:
If we take this example below and run it with your branch vs. master, we see there is a different behavior when running into the second epoch. 
Let's think about how to rectify this tomorrow


```py

import os
import time

import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        time.sleep(0.5)
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        time.sleep(0.5)
        loss = self(batch).sum()
        self.log("valid_loss", loss)

    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 10), batch_size=1)
    val_data = DataLoader(RandomDataset(32, 5), batch_size=1)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        num_sanity_val_steps=0,
        max_epochs=2,
        val_check_interval=0.5
    )
    trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data)


if __name__ == "__main__":
    run()

```

Reply:
I wondered if somebody would notice this :P

I opened this PR at the same time than #11244. That one fixes the bug you noticed. This branch wasn't updated after #11244 got merged.

You'll see it working properly now that I've merged master.

Reply:
Yes seems to work now!